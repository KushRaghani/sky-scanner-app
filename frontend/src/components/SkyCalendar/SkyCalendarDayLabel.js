import React from "react";
import { getDaysOfWeek } from "../../utils/dateUtil";

const SkyCalendarDayLabel = () => {
  const daysOfWeek = getDaysOfWeek();
  // Iterate over days of the week and create labels (Sunday to Saturday)
  return (
    <div className="calendar-days calendar-row">
      {daysOfWeek.map((item, i) => {
        return (
          <div className="block block-center" key={i}>
            {item}
          </div>
        );
      })}
    </div>
  );
};

export default SkyCalendarDayLabel;
