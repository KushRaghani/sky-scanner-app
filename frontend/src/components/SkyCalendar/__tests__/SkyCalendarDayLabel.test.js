import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import SkyCalendarDayLabel from "../SkyCalendarDayLabel";

function loadSkyCalendarDayLabel() {
  return <SkyCalendarDayLabel />;
}
describe("SkyCalendarDayLabel", () => {
  // Enyzme tests
  test("renders SkyCalendarDayLabel", () => {
    const wrapper = shallow(loadSkyCalendarDayLabel());
    expect(wrapper.find(".block.block-center").length).toBe(7);
    expect(wrapper.find(".calendar-days.calendar-row").length).toBe(1);
    expect(
      wrapper
        .find(".block.block-center")
        .at(0)
        .text()
    ).toEqual("Sunday");
    expect(
      wrapper
        .find(".block.block-center")
        .at(1)
        .text()
    ).toEqual("Monday");
    expect(
      wrapper
        .find(".block.block-center")
        .at(2)
        .text()
    ).toEqual("Tuesday");
    expect(
      wrapper
        .find(".block.block-center")
        .at(3)
        .text()
    ).toEqual("Wednesday");
    expect(
      wrapper
        .find(".block.block-center")
        .at(4)
        .text()
    ).toEqual("Thursday");
    expect(
      wrapper
        .find(".block.block-center")
        .at(5)
        .text()
    ).toEqual("Friday");
    expect(
      wrapper
        .find(".block.block-center")
        .at(6)
        .text()
    ).toEqual("Saturday");
  });

  // Snapshot testing
  test("snapshot with no data", () => {
    const tree = renderer.create(loadSkyCalendarDayLabel());
    expect(tree).toMatchSnapshot();
  });
});
