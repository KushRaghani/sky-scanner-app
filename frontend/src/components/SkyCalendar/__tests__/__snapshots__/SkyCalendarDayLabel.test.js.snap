// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkyCalendarDayLabel snapshot with no data 1`] = `
<div
  className="calendar-days calendar-row"
>
  <div
    className="block block-center"
  >
    Sunday
  </div>
  <div
    className="block block-center"
  >
    Monday
  </div>
  <div
    className="block block-center"
  >
    Tuesday
  </div>
  <div
    className="block block-center"
  >
    Wednesday
  </div>
  <div
    className="block block-center"
  >
    Thursday
  </div>
  <div
    className="block block-center"
  >
    Friday
  </div>
  <div
    className="block block-center"
  >
    Saturday
  </div>
</div>
`;
